docs/core/v3.0.rst

```diff
@@ -0,0 +1,1762 @@
+define a set of core features that are useful and sufficient to
+address a significant fraction of use cases, but are also
+straightforward to implement fully across different programming
+languages. Additional functionality can then be layered via
+extensions, some of which may aim for wide adoption, some of which may
+be more specialised and have more limited implementation.
+
+
+Extensibility
+-------------
+
+The development of systems for storage of very large array-like data
+is a very active area of research and development, and there are many
+possibilities that remain to be explored. A goal of this specification
+is to define a format with a number of clear extension points and
+mechanisms, in order to provide a framework for freely building on and
+exploring these possibilities. We aim to make this possible, whilst
+also providing pathways for a graceful degradation of functionality
+where possible, in order to retain interoperability. We also aim to
+provide a framework for community-defined extensions, which can be
+developed and published independently without requiring centralised
+coordination of all specifications.
+
+See `extensions`_ below.
+
+Stability Policy
+----------------
+
+This core specification adheres to a ``MAJOR.MINOR`` version
+number format. A zarr implementation provides the read and write API by
+implementing this specification can be considered compatible with all
+datasets following the specification with the same major version number.
+
+Notably, this excludes extensions, codecs and stores from the compatibility
+of the core specification. However, versioned extensions and stores are also
+expected to follow this stability policy.
+
+This means that implementations based on a ``3.X`` specification will be
+able to read and write to datasets that follow any ``3.Y`` specification,
+as long as it only uses
+
+- optional extensions or those supported by the implementation,
+- no codec or one supported by the implementation, and
+- the used store must be supported by the implementation.
+
+For details, please see the `zarr_format`_ metadata entry.
+
+
+Questions that still need to be resolved
+----------------------------------------
+
+We solicit feedback on the following area during the RFC period of this first
+draft.
+
+ - Should core metadata and user attributes be stored together or separate documents?
+   (See https://github.com/zarr-developers/zarr-specs/issues/72)
+ - extensions and ``must_understand = True`` might be too restrictive. Work a
+   We propose to develop a draft implementation with extensions and
+   see how far we can go. A possible list of extensions to include:
+
+    - Boolean
+    - Complex
+    - Datetime
+    - Named dimensions
+    - Awkward arrays
+
+   See https://github.com/zarr-developers/zarr-specs/issues/89 for discussion on
+   the topic.
+
+  - Node name case sensitivity: The node name is now case sensitive. This may
```

Comment:
On Windows and macOS the filesystem is case insensitive (but case preserving).  I think in practice users will want to be able to use a "directory store" (without additional escaping) on those operating systems.  Therefore I think it might be better to explicitly note that some implementations may not support two nodes with names that differ only in case.

Reply:
Added notes about this in 41db4a3 of #175